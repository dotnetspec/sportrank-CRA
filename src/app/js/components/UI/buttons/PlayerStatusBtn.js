import React from 'react';
import { Button } from 'react-bootstrap';
import JSONops from '../../Logic/JSONops';
/**
 * Functional component representing player status button in the header.
 * The component rendering in this area is controlled by
 * an @external "BrowserRouter"
 *
 * @extends React.Component
 */
 const PlayerStatusBtn = (username) => {

   const [playerActive, setPlayerActive] = React.useState(true);
   const [btnText, setValue] = React.useState('De-Activate?');
   const [bsStyle, setStyle] = React.useState('success');


   //const onChange = event => setValue(event.target.value);
     const _handleChangeStatusPlayerBtnText = (username, props) => {
     // const _handleChangeStatusPlayerBtnText = event =>
     // {
       if(username !== null){
           if(playerActive === true){
             setPlayerActive(false);
             setStyle('warning');
             setValue('Re-Activate?');
           }else{
             setPlayerActive(true );
             setStyle('success');
             setValue('De-Activate?')
             try {
               console.log('in _handleReactivatePlayer', props.newrankIdCB, this.props.rankingJSONdata, username, this.props.account)
               JSONops.reactivatePlayer(this.props.newrankIdCB, this.props.rankingJSONdata, username, this.props.account);
               this.props.history.push('/home/@' + username);
             } catch (err) {
             // stop loading state and show the error
             console.log(err.message);
             };
           }
         }else {
             console.log('no username passed to deactivate btn!')
           }
         }


   return (
       <Button className='deactivatebtn'  data-cy='deactivate'
       placeholder="De-Activate?"
       data-testid="activatebtn-input"
       bsStyle={bsStyle}
       onClick={(e) => _handleChangeStatusPlayerBtnText(username)}
       >
         {btnText}
       </Button>
   );
 };


 export default PlayerStatusBtn
