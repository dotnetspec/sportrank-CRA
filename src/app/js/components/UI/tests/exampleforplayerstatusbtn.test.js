// src/__tests__/example.js
// query utilities:
import {
  getByLabelText,
  getByText,
  getByTestId,
  queryByTestId,
  // Tip: all queries are also exposed on an object
  // called "queries" which you could import here as well
  wait,
} from '@testing-library/dom'
// adds special assertions like toHaveTextContent
import 'jest-dom/extend-expect'
import PlayerStatusBtn from '../buttons/PlayerStatusBtn'
import React from 'react'
import { render } from '@testing-library/react'



function getExampleDOM() {
  // This is just a raw example of setting up some DOM
  // that we can interact with. Swap this with your UI
  // framework of choice ðŸ˜‰
  const div = document.createElement('div')
  div.innerHTML = `
    <label for="username">Username</label>
    <input id="username" />
    <button>Print Username</button>
  `
  const button = div.querySelector('button')
  const input = div.querySelector('input')
  button.addEventListener('click', () => {
    // let's pretend this is making a server request, so it's async
    // (you'd want to mock this imaginary request in your unit tests)...
    setTimeout(() => {
      const printedUsernameContainer = document.createElement('div')
      printedUsernameContainer.innerHTML = `
        <div data-testid="printed-username">${input.value}</div>
      `
      div.appendChild(printedUsernameContainer)
    }, Math.floor(Math.random() * 200))
  })
  return div
}

// function getExampleDOMFromReact() {
//   // This is just a raw example of setting up some DOM
//   // that we can interact with. Swap this with your UI
//   // framework of choice ðŸ˜‰
//   const div = document.createElement('div')
//   div.innerHTML = <PlayerStatusBtn />
//   return div
// }

test('examples of some things', async () => {
  const playerStatusBtnText = 'Re-Activate?'
  //const container = getExampleDOMFromReact()

  // Get form elements by their label text.
  // An error will be thrown if one cannot be found (accessibility FTW!)
  //const input = getByLabelText(container, 'dereaactivatebtn')
  //const { getByLabelText, queryByTestId,  getByTestId} = render(<PlayerStatusBtn />)
  //const inputNode = getByLabelText('dereaactivatebtn')

  const historyMock = { push: jest.fn() };

  const { getByPlaceholderText, getByTestId} = render(<PlayerStatusBtn username='player1'
  history={historyMock}
  />)
  const inputNode = getByPlaceholderText('De-Activate?')

  inputNode.btnText = playerStatusBtnText

  // Get elements by their text, just like a real user does.
  //getByText(container, 'De-Activate?').click()
  inputNode.click();

  await wait(() =>
    //expect(queryByTestId(container, 'printed-username')).toBeTruthy()
    //expect(queryByTestId('printed-username')).toBeTruthy()
    expect(getByTestId('activatebtn-input')).toHaveTextContent(
      playerStatusBtnText
    )
  )

  // getByTestId and queryByTestId are an escape hatch to get elements
  // by a test id (could also attempt to get this element by it's text)

  // jest snapshots work great with regular DOM nodes!
  expect(inputNode).toMatchSnapshot()
})
