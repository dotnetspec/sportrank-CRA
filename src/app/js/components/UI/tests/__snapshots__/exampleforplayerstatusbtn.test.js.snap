// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples of some things 1`] = `
<button
  class="deactivatebtn btn btn-warning"
  data-cy="deactivate"
  data-testid="activatebtn-input"
  placeholder="De-Activate?"
  type="button"
>
  Re-Activate?
</button>
`;
